src/od/process/mod.rs

```diff
@@ -736,15 +744,15 @@ where
                 Err(e) => return Err(e),
             }
             self.prop.state.reset_stm();
-            if epoch == end_epoch {
+            if epoch >= end_epoch {
```

Comment:
This new approach may map the covariance a fraction of a step further than requested.